packages/app-desktop/gui/ResizableLayout/utils/useLayoutItemSizes.test.ts

```diff
@@ -138,4 +138,157 @@ describe('useLayoutItemSizes', () => {
 		expect(itemSize(parent.children[1], parent, sizes, false)).toEqual({ width: 95, height: 50 });
 	});
 
+	test('should decrease size of the largest item if the total size would be larger than the container', () => {
+		const layout: LayoutItem = validateLayout({
+			key: 'root',
+			width: 200,
+			height: 100,
+			direction: LayoutItemDirection.Row,
+			children: [
+				{
+					key: 'col1',
+					width: 110,
+				},
+				{
+					key: 'col2',
+					width: 100,
+				},
+				{
+					key: 'col3',
+					minWidth: 50,
+				},
+			],
+		});
+
+		const { result } = renderHook(() => useLayoutItemSizes(layout));
+		const sizes = result.current;
+
+		expect(sizes.col1.width).toBe(50);
+		expect(sizes.col2.width).toBe(100);
+	});
+
+	test('should ignore invisible items when counting remaining size', () => {
+		const layout: LayoutItem = validateLayout({
+			key: 'root',
+			width: 200,
+			height: 100,
+			direction: LayoutItemDirection.Row,
+			children: [
+				{
+					key: 'col1',
+					width: 110,
+					visible: false,
+				},
+				{
+					key: 'col2',
+					width: 100,
+				},
+				{
+					key: 'col3',
+					minWidth: 50,
+				},
+			],
+		});
+
+		const { result } = renderHook(() => useLayoutItemSizes(layout));
+		const sizes = result.current;
+
+		expect(sizes.col2.width).toBe(100);
+	});
+
+	test('should ignore invisible items when selecting largest child', () => {
+		const layout: LayoutItem = validateLayout({
+			key: 'root',
+			width: 200,
+			height: 100,
+			direction: LayoutItemDirection.Row,
+			children: [
+				{
+					key: 'col1',
+					width: 110,
+					visible: false,
+				},
+				{
+					key: 'col2',
+					width: 100,
+				},
+				{
+					key: 'col3',
+					width: 110,
+				},
+				{
+					key: 'col4',
+					minWidth: 50,
+				},
+			],
+		});
+
+		const { result } = renderHook(() => useLayoutItemSizes(layout));
+		const sizes = result.current;
+
+		expect(sizes.col2.width).toBe(100);
+		expect(sizes.col3.width).toBe(50);
+	});
+
+});
+
+describe('calculateMaxSizeAvailableForItem', () => {
+
+	test('should give maximum available space this item can take up during resizing', () => {
+		const layout: LayoutItem = validateLayout({
+			key: 'root',
+			width: 200,
+			height: 100,
+			direction: LayoutItemDirection.Row,
+			children: [
+				{
+					key: 'col1',
+					width: 50,
+				},
+				{
+					key: 'col2',
+					width: 70,
+				},
+				{
+					key: 'col3',
+				},
+			],
+		});
+
+		const { result } = renderHook(() => useLayoutItemSizes(layout));
+		const sizes = result.current;
+		const maxSize = calculateMaxSizeAvailableForItem(layout.children[0], layout, sizes);
+
+		expect(maxSize.width).toBe(90);
```

Comment:
Again I think it would be best to test all the columns. I understand the calculation but that's because I know the min width is 40 (something I'll forget by next week :). Or perhaps add a comment that explains the reasoning (or both)?